enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,1169 @@
+
+  // networks is the list of networks within this failure domain
+  // +kubebuilder:validation:Optional
+  Networks []string `json:"networks,omitempty"`
+
+  // datastore is the name or inventory path of the datastore in which the
+  // virtual machine is created/located.
+  // TODO: verify maximum length
+  // TODO: what if this is a inventory path?
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=256
+  Datastore string `json:"datastore,omitempty"`
+}
+
+type VSpherePlatformFailureDomainHosts struct {
+  // VMGroupName is the name of the VM group
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength=1024
+  VMGroupName string `json:"vmGroupName"`
+
+  // HostGroupName is the name of the Host group
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength=1024
+  HostGroupName string `json:"hostGroupName"`
+}
+
+// VSpherePlatformVCenterSpec stores the vCenter connection fields
+// https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go
+type VSpherePlatformVCenterSpec struct {
+
+  // server is the fully-qualified domain name or the IP address of the vCenter server.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength=255
+  Server string `json:"server"`
+
+  // port is the TCP port that will be used to communicate to
+  // the vCenter endpoint. This is typically unchanged from
+  // the default of HTTPS TCP/443.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Minimum=1
+  // +kubebuilder:validation:Maximum=32767
+  // +kubebuilder:default=443
+  Port uint `json:"port,omitempty"`
+
+  // insecureFlag is a boolean, if set to true
+  // will disable certificate verification and
+  // set Go's `tls.Config.InsecureSkipVerify` to true.
+  // > If `InsecureSkipVerify` is true, TLS accepts any certificate
+  // > presented by the server and any host name in that certificate.
+  // Setting this value to false will require a valid certificate
+  // or the self-signed to be added to the certificate store.
+  // +kubebuilder:validation:Optional
+  InsecureFlag bool `json:"insecureFlag,omitempty"`
+
+  // Datacenter in which VMs are located.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinItems=1
+  Datacenters []string `json:"datacenters"`
+
+  // Soap round tripper count (retries = RoundTripper - 1)
+  // +kubebuilder:validation:Optional
+  RoundTripperCount uint `json:"soapRoundtripCount,omitempty"`
+
+  // Specifies the path to a CA certificate in PEM format. Optional; if not
+  // configured, the system's CA certificates will be used.
+  // +kubebuilder:validation:Optional
+  CAFile string `json:"caFile,omitempty"`
+
+  // Thumbprint of the VCenter's certificate thumbprint
+  // +kubebuilder:validation:Optional
+  Thumbprint string `json:"thumbprint,omitempty"`
```

Comment:
Would be nice to know a bit more about the expectations here, is this part of the tls verification, what happens if you rotate the cert? Why is CAFile not enough? Are any other options required?